torch/_inductor/fx_passes/post_grad.py

```diff
@@ -816,9 +816,8 @@ def scatter_upon_const_tensor(
 
     # Check if inputs are tensors instead of inductor IR nodes
     if isinstance(selector, torch.Tensor):
-        # Return a fake tensor with the proper shape that this operator is intended to return
-        device = selector.device if hasattr(selector, "device") else torch.device("cpu")
-        return torch.empty(shape, dtype=dtype, device=device)
+        background_tensor = torch.full(shape, background_val, dtype=dtype)
+        return torch.scatter(background_tensor, dim, selector, val)
```

Comment:
the previous implementation caused a numerical issue in https://github.com/pytorch/torchtitan/pull/1785